hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/solr/SolrUtil.java

```diff
@@ -0,0 +1,492 @@
+      volumeInfo.setSize(volumeInfo.getSize() + bucket.getSize());
+      updateBucketLevelMinMaxAccessCount(bucket);
+      updateBucketColors(bucket);
+    });
+  }
+
+  private void updateBucketColors(EntityReadAccessHeatMapResponse bucket) {
+    long delta = bucket.getMaxAccessCount() - bucket.getMinAccessCount();
+    List<EntityReadAccessHeatMapResponse> children =
+        bucket.getChildren();
+    children.stream().forEach(path -> {
+      path.setColor(1.000);
+      if (delta > 0) {
+        double truncatedValue = truncate(
+            ((double) path.getAccessCount() /
+                (double) bucket.getMaxAccessCount()), 3);
+        path.setColor(truncatedValue);
+      }
+    });
+  }
+
+  private static double truncate(double value, int decimalPlaces) {
+    if (decimalPlaces < 0) {
+      throw new IllegalArgumentException();
+    }
+    value = value * Math.pow(10, decimalPlaces);
+    value = Math.floor(value);
+    value = value / Math.pow(10, decimalPlaces);
+    return value;
+  }
+
+  private void updateRootEntitySize(
+      EntityReadAccessHeatMapResponse rootEntity) {
+    List<EntityReadAccessHeatMapResponse> children =
+        rootEntity.getChildren();
+    children.stream().forEach(volume -> {
+      updateVolumeSize(volume);
+      rootEntity.setSize(rootEntity.getSize() + volume.getSize());
+    });
+  }
+
+  private void addBucketAndPrefixPath(
+      String[] split, EntityReadAccessHeatMapResponse volumeEntity,
+      long readAccessCount, long keySize) {
+    List<EntityReadAccessHeatMapResponse> bucketEntities =
+        volumeEntity.getChildren();
+    EntityReadAccessHeatMapResponse bucket =
+        new EntityReadAccessHeatMapResponse();
+    bucket.setLabel(split[1]);
+    bucketEntities.add(bucket);
+    bucket.setMinAccessCount(readAccessCount);
+    addPrefixPathInfoToBucket(split, bucket, readAccessCount, keySize);
+  }
+
+  private void addPrefixPathInfoToBucket(
+      String[] split, EntityReadAccessHeatMapResponse bucket,
+      long readAccessCount, long keySize) {
+    List<EntityReadAccessHeatMapResponse> prefixes = bucket.getChildren();
+    updateBucketSize(bucket, keySize);
+    String path = Arrays.stream(split)
+        .skip(2).collect(Collectors.joining("/"));
+    EntityReadAccessHeatMapResponse prefixPathInfo =
+        new EntityReadAccessHeatMapResponse();
+    prefixPathInfo.setLabel(path);
+    prefixPathInfo.setAccessCount(readAccessCount);
+    prefixPathInfo.setSize(keySize);
+    prefixes.add(prefixPathInfo);
+    // This is done for specific ask by UI treemap to render and provide
+    // varying color shades based on varying ranges of access count.
+    updateRootLevelMinMaxAccessCount(readAccessCount);
+  }
+
+  private void updateBucketLevelMinMaxAccessCount(
+      EntityReadAccessHeatMapResponse bucket) {
+    List<EntityReadAccessHeatMapResponse> children =
+        bucket.getChildren();
+    bucket.setMinAccessCount(Long.MAX_VALUE);
```

Comment:
default is minAccess and maxAccess count is "0", just considering from this code, if children count is "0", minAccess Count remains MAX_VALUE

Reply:
Ok, this is fixed.